# Table of content

1. Importing libraries

2. Importing data

        Importing prior orders dataset
        Importing orders dataset

3. Merging orders and prior orders datasets

4. Exporting data in pickle format

# 01. Importing Libraries

In [1]:
# Importing libraries
import pandas as pd
import numpy as np
import os

# 02. Importing Data

In [2]:
# Creating shortcut path
path = path = r'C:\Users\henri\Documents\Tonia\Learning\CarrrerFoundry\Data Immersion\Achievement 4\Instacart Basket Analysis'

In [3]:
path

'C:\\Users\\henri\\Documents\\Tonia\\Learning\\CarrrerFoundry\\Data Immersion\\Achievement 4\\Instacart Basket Analysis'

## Importing prior orders dataset

In [4]:
# Importing df_ords_prior.csv
df_ords_prior = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'order_products__prior.csv'), index_col = False)

In [5]:
df_ords_prior.head()

order_id  product_id  add_to_cart_order  reordered
0         2       33120                  1          1
1         2       28985                  2          1
2         2        9327                  3          0
3         2       45918                  4          1
4         2       30035                  5          0

In [6]:
df_ords_prior.shape

(32434489, 4)

In [18]:
# Checking the data types
df_ords_prior.dtypes

order_id             int64
product_id           int64
add_to_cart_order    int64
reordered            int64
dtype: object

In [19]:
# Changing "order_id" data type in prior orders dataframe
df_ords_prior['order_id'] = df_ords_prior['order_id'].astype('str')

In [20]:
# Checking the data types
df_ords_prior.dtypes

order_id             object
product_id            int64
add_to_cart_order     int64
reordered             int64
dtype: object

## Importing orders dataset

In [7]:
# Importing orders_clean.csv
df_ords = pd.read_csv(os.path.join(path, '02 Data', 'Prepared Data', 'orders_clean.csv'), index_col = False)

In [8]:
df_ords.head()

Unnamed: 0  order_id  user_id  order_number  orders_day_of_week  \
0           0   2539329        1             1                   2   
1           1   2398795        1             2                   3   
2           2    473747        1             3                   3   
3           3   2254736        1             4                   4   
4           4    431534        1             5                   4   

   order_hour_of_day  days_since_previous_order  
0                  8                        NaN  
1                  7                       15.0  
2                 12                       21.0  
3                  7                       29.0  
4                 15                       28.0

In [9]:
df_ords.dtypes

Unnamed: 0                     int64
order_id                       int64
user_id                        int64
order_number                   int64
orders_day_of_week             int64
order_hour_of_day              int64
days_since_previous_order    float64
dtype: object

In [10]:
# Dropping column "Unnamed":
df_ords = df_ords.drop(columns = ['Unnamed: 0'])

In [11]:
df_ords.head()

order_id  user_id  order_number  orders_day_of_week  order_hour_of_day  \
0   2539329        1             1                   2                  8   
1   2398795        1             2                   3                  7   
2    473747        1             3                   3                 12   
3   2254736        1             4                   4                  7   
4    431534        1             5                   4                 15   

   days_since_previous_order  
0                        NaN  
1                       15.0  
2                       21.0  
3                       29.0  
4                       28.0

In [12]:
# Changing "order_id" and "user_id" data type in orders dataframe
df_ords['order_id'] = df_ords['order_id'].astype('str')
df_ords['user_id'] = df_ords['user_id'].astype('str')

In [13]:
df_ords.dtypes

order_id                      object
user_id                       object
order_number                   int64
orders_day_of_week             int64
order_hour_of_day              int64
days_since_previous_order    float64
dtype: object

In [14]:
df_ords.shape

(3421083, 6)

# 03. Merging orders and prior orders dataframes

In [21]:
# Creating a new dataframe, df_merged_large
df_merged_large = df_ords.merge(df_ords_prior, on = 'order_id', indicator = True)

In [22]:
# First rows of newly merged dataframe
df_merged_large.head()

order_id user_id  order_number  orders_day_of_week  order_hour_of_day  \
0  2539329       1             1                   2                  8   
1  2539329       1             1                   2                  8   
2  2539329       1             1                   2                  8   
3  2539329       1             1                   2                  8   
4  2539329       1             1                   2                  8   

   days_since_previous_order  product_id  add_to_cart_order  reordered _merge  
0                        NaN         196                  1          0   both  
1                        NaN       14084                  2          0   both  
2                        NaN       12427                  3          0   both  
3                        NaN       26088                  4          0   both  
4                        NaN       26405                  5          0   both

In [24]:
# Check to see if we have a full match or not
df_merged_large['_merge'].value_counts()

both          32434489
left_only            0
right_only           0
Name: _merge, dtype: int64

Note: the resulting dataframe (after the merge) has 32,434,489 rows, and each of those rows have information found in both input data sets.

# 04. Exporting Data in Pickle Format

In [25]:
# Exporting merged dataframe to pickle
df_merged_large.to_pickle(os.path.join(path, '02 Data', 'Prepared Data', 'orders_products_combined.pkl'))